tests/rt.sh

```diff
@@ -787,16 +807,22 @@ elif [[ $MACHINE_ID = jet ]]; then
 
 elif [[ $MACHINE_ID = s4 ]]; then
 
-  module load rocoto/1.3.2
-  module load ecflow/5.6.0
+  if [[ "${ROCOTO:-false}" == true ]] ; then
+    module load rocoto/1.3.2
+  fi
+  if [[ "${ECFLOW:-false}" == true ]] ; then
```

Comment:
This should be elif. -r an -e won't be used at the same time.

Reply:
To make that consistent across all the machines, I'd have a lot of "ifs" and "module loads" to rearrange. It would change the order of module loads.

Reply:
If we are adding conditional loading of rocoto and ecflow modules then all rocoto and ecflow variables should be moved in the corresponding if blocks. For example, in this section:
```
  module use /ncrc/proj/epic/rocoto/modulefiles
  if [[ "${ROCOTO:-false}" == true ]] ; then
    module load rocoto
  fi
  ROCOTORUN=$(which rocotorun)
  ROCOTOSTAT=$(which rocotostat)
  ROCOTOCOMPLETE=$(which rocotocomplete)
```
ROCOTORUN, ROCOTOSTAT and ROCOTOCOMPLETE should not be defined outside the 'if rocoto' block. Also, 'module use ....' should be moved.

Reply:
You're right. All I did was a global editor macro execution for the "module load". I'll go back through and do what you suggest.

Reply:
I did it. Please read through and tell me if it looks correct.
This is a pretty extensive change though. I'm retesting Rocoto on Hercules and Hera to make sure it works. Only a few regression tests; not the whole suite.

Reply:
My mini-test of a few regression tests with Rocoto passed on Hercules and Hera.

Reply:
Out of paranoia, I'm testing on Jet and Orion too.

Reply:
Jet and Orion mini-tests passed now.